pennylane/devices/default_qubit.py

```diff
@@ -713,7 +784,13 @@ def _apply_unitary_einsum(self, state, mat, wires):
         # translate to wire labels used by device
         device_wires = self.map_wires(wires)
 
-        mat = self._cast(self._reshape(mat, [2] * len(device_wires) * 2), dtype=self.C_DTYPE)
+        dim = 2 ** len(device_wires)
+        batch_size = self._get_batch_size(mat, (dim, dim), dim**2)
+
+        shape = [2] * (len(device_wires) * 2)
+        if batch_size is not None:
+            shape.insert(0, batch_size)
+        mat = self._cast(self._reshape(mat, shape), dtype=self.C_DTYPE)
```

Comment:
If the matrix is broadcasted, it again is reshaped to a shape including the broadcasting dim as leading dim.